warehouse/ingest-core/src/main/java/datawave/iterators/FrequencyColumnIterator.java

```diff
@@ -0,0 +1,119 @@
+package datawave.iterators;
+
+import datawave.data.ColumnFamilyConstants;
+import datawave.query.util.FrequencyFamilyCounter;
+import datawave.query.util.MetadataHelper;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.user.TransformingIterator;
+import org.apache.hadoop.io.Text;
+import java.io.IOException;
+
+public class FrequencyColumnIterator extends TransformingIterator {
+    
+    private FrequencyFamilyCounter frequencyFamilyCounter;
+    
+    public FrequencyColumnIterator() {};
+    
+    public FrequencyColumnIterator(FrequencyColumnIterator aThis, IteratorEnvironment environment) {
+        super();
+        setSource(aThis.getSource().deepCopy(environment));
+    }
+    
+    @Override
+    protected PartialKey getKeyPrefix() {
+        return PartialKey.ROW_COLFAM;
+    }
+    
+    @Override
+    protected void transformRange(SortedKeyValueIterator<Key,Value> sortedKeyValueIterator, KVBuffer kvBuffer) throws IOException {
+        Key newKey = null;
+        Long numRecords = 0L;
+        frequencyFamilyCounter = new FrequencyFamilyCounter();
+        
+        log.trace("Transforming range for key " + sortedKeyValueIterator.getTopKey().getRow().toString(), new Exception());
+        
+        Key topKey = null, aggregatedKey = null;
+        Value topValue = null;
+        Value aggregatedValue = null;
+        boolean isFrequencyRange = false;
+        
+        while (sortedKeyValueIterator.hasTop()) {
+            numRecords++;
+            Text cq = sortedKeyValueIterator.getTopKey().getColumnQualifier();
+            topKey = sortedKeyValueIterator.getTopKey();
+            topValue = sortedKeyValueIterator.getTopValue();
+            
+            // Just do an identity transform if there it is not a Frequency Column
+            if (!topKey.getColumnFamily().toString().equals(ColumnFamilyConstants.COLF_F.toString())) {
+                kvBuffer.append(sortedKeyValueIterator.getTopKey(), sortedKeyValueIterator.getTopValue());
+                sortedKeyValueIterator.next();
+                isFrequencyRange = false;
+                continue;
+            } else {
+                isFrequencyRange = true;
+            }
+            
+            if (cq.toString().startsWith(MetadataHelper.COL_QUAL_PREFIX)) {
+                newKey = topKey;
+                frequencyFamilyCounter.deserializeCompressedValue(topValue);
```

Comment:
Is it not possible to have multiple compressed values for the same field in which case this needs to actually aggregate.  We will have multiple compressed values if we do a merging major compaction (only triggered when you have a large number of table files) or from minor compactions.  This may be one of the reasons you couldn't get this iterator on minor compactions.